### PR TITLE
feat(settings): make the settings page remember where you were

### PR DESCRIPTION
**Pull request by sdwoodbury**
### What this PR does 📖

- go to the settings page, should init to `Profile`.
- click on a different settings page.
- leave the settings page
- return to the settings page - should remember where you left off. 

### Which issue(s) this PR fixes 🔨

- Resolve #

### CODE REVIEW

**Comment by luisecm** [comment]
~~I noticed that a minor change would be required on Windows Chats - Quick Profile tests (test is looking for Settings Profile, but last Settings screen for the test user is Setting Notifications). To fix this, I would need to add one line in test to go from Settings Notifications to Settings Profile. Short comment is that chats tests will fail here~~

Update: Please look at Phil's comment below:

**Comment by phillsatellite** [comment]
Along with what Luis had mentioned (might be a little different than what he is describing) If Im in Settings and go to say the About Page. Then I go to Chat/QuickProfile/EditProfile instead of being taken to the Edit Profile, I land on the About Page. Other than that I think this pr is awesome 👍

**Review by phillsatellite — APPROVED** [review]

File: ui/src/components/settings/sidebar.rs [hunk 1]
```diff
@@ -12,6 +12,7 @@ use kit::{
     layout::sidebar::Sidebar as ReusableSidebar,
 };
 
+#[derive(Clone, Eq, PartialEq)]
```
Comment (dariusc93) [inline ui/src/components/settings/sidebar.rs]:
```suggestion
#[derive(Copy, Clone, Eq, PartialEq)]
```

File: ui/src/components/settings/sidebar.rs [hunk 2]
```diff
@@ -26,6 +27,22 @@ pub enum Page {
     Licenses,
 }
 
+impl Page {
+    pub fn set(&mut self, p: Page) {
+        *self = p;
+    }
+    pub fn get(&self) -> Self {
+        self.clone()
```
Comment (dariusc93) [inline ui/src/components/settings/sidebar.rs]:
```suggestion
        *self
```

**Review by dariusc93 — APPROVED** [review]
Some minor suggestions but not blocking to the PR

**Review by stavares843 — APPROVED** [review]

**Comment by stavares843** [comment]
cc @sdwoodbury regarding darius feedback above

**Pull request merged**